### PR TITLE
dropAll() return immediately if no tables found

### PR DESCRIPTION
**Pull request by matthewjumpsoffbuildings**
Small tweak to Schema Builder dropAll() - if no tables exists simply return immediately

### CODE REVIEW

**Comment by taka-oyama** [comment]
Please add a test that runs this code.

**Comment by matthewjumpsoffbuildings** [comment]
Im actually not sure how to test this? Since dropAll() returns void, how would I test if it exited early due to there being no tables, or if it exited later because it found and dropped some tables?

**Comment by taka-oyama** [comment]
I'm ok as long as all the code paths are covered.
So you can just run Connection::dropAllTables with no tables and just `$this->assertTrue(true);`.

**Comment by matthewjumpsoffbuildings** [comment]
added test_dropAllTablesEmpty(), that should cover it?

**Comment by matthewjumpsoffbuildings** [comment]
It appears to work when I run it, but I am not sure about a couple of things

Does that test_dropAllTablesEmpty() get run last in the file? If so it will be okay since the previous test_dropAllTables() will have emptied the db. Or do the test methods in the file get called in any order, in which case does each test method start with an empty database, or do artefacts from previous tests persist?

**Comment by taka-oyama** [comment]
I think tests that depend on other tests to be run first is really bad.

So ideally, there should be a hook which deletes all created tables after every test, but that slows down the test significantly.
So as a compromise, I think you should drop all tables twice in this test. That way the 2nd try will always go through the code provided.

**Comment by matthewjumpsoffbuildings** [comment]
Done

File: tests/Schema/BuilderTestLast.php [hunk 1]
```diff
@@ -307,4 +307,19 @@ public function test_dropAllTables(): void
         $tables = $sb->getTables();
         $this->assertEmpty($tables);
     }
+
+    public function test_dropAllTablesEmpty(): void
+    {
+        $conn = $this->getDefaultConnection();
+        $sb = $conn->getSchemaBuilder();
+        $sb->dropAllTables();
```
Comment (taka-oyama) [inline tests/Schema/BuilderTestLast.php:315]:
```suggestion

         // All tables must be dropped before hand to ensure that this test will cover the "early return" case
         // for when there are no tables.
        $sb->dropAllTables();
```

File: tests/Schema/BuilderTestLast.php [hunk 2]
```diff
@@ -307,4 +307,22 @@ public function test_dropAllTables(): void
         $tables = $sb->getTables();
         $this->assertEmpty($tables);
     }
+
+    public function test_dropAllTablesEmpty(): void
```
Comment (taka-oyama) [inline tests/Schema/BuilderTestLast.php:311]:
```suggestion
    public function test_dropAllTables_when_no_tables_exist(): void
```

**Review by taka-oyama — APPROVED** [review]

**Comment by taka-oyama** [comment]
Please add a changelog

**Comment by matthewjumpsoffbuildings** [comment]
Updated the changelog, not sure if its a new point release, please edit if needed

Also should we add #191 and #192 to the changelog too?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,8 @@
+# v7.0.1? (?)
```
Comment (taka-oyama) [inline CHANGELOG.md:1]:
```suggestion
# v7.1.0 (Not released yet)
```

**Comment by taka-oyama** [comment]
> Also should we add https://github.com/colopl/laravel-spanner/pull/191 and https://github.com/colopl/laravel-spanner/pull/192 to the changelog too?

Yes but I'd like to have a different PR for them.

**Review by taka-oyama — APPROVED** [review]

**Comment by matthewjumpsoffbuildings** [comment]
> Yes but I'd like to have a different PR for them.

Do you want me to do anything or you got it?

**Comment by taka-oyama** [comment]
I'd appreciate it if you can write the ones for the previous two. Thanks.

**Comment by matthewjumpsoffbuildings** [comment]
Which branch should I do it on?

**Comment by taka-oyama** [comment]
master for all unreleased features.

**Pull request merged**